File: frontend/src/types/media.ts
export interface AudioTrack {
  id: number
  stream_index: number
  language: string | null
  language_code: string | null
  codec: string | null
  channels: string | null
  bitrate: string | null
  is_default: boolean
  title: string | null
}

export interface SubtitleTrack {
  id: number
  stream_index: number
  language: string | null
  language_code: string | null
  format: string | null
  forced: boolean
  is_default: boolean
  title: string | null
}

export interface MediaItem {
  id: number
  title: string
  year: number | null
  media_type: 'movie' | 'show'
  rating: string | null
  poster_url: string | null
  quality: string | null
  size: string | null
  size_bytes: number
  runtime: string | null
  video_codec: string | null
  container: string | null
  audio_track_count: number
  subtitle_track_count: number
}

export interface MediaDetail {
  id: number
  title: string
  year: number | null
  media_type: 'movie' | 'show'
  rating: string | null
  poster_url: string | null
  quality: string | null
  size: string | null
  size_bytes: number
  runtime: string | null
  video_codec: string | null
  container: string | null
  file_path: string | null
  arr_id: number | null
  arr_type: string | null
  last_scanned: string | null
  audio_tracks: AudioTrack[]
  subtitle_tracks: SubtitleTrack[]
}

export interface LibraryStats {
  total_media: number
  total_movies: number
  total_shows: number
  total_audio_tracks: number
  total_subtitle_tracks: number
  total_size_gb: number
}

File: backend/app/schemas/media.py
"""Pydantic schemas for media API responses."""

from pydantic import BaseModel


class AudioTrackResponse(BaseModel):
    stream_index: int
    language: str | None
    language_code: str | None
    codec: str | None
    channels: str | None
    bitrate: str | None
    is_default: bool
    title: str | None


class SubtitleTrackResponse(BaseModel):
    stream_index: int
    language: str | None
    language_code: str | None
    format: str | None
    forced: bool
    is_default: bool
    title: str | None


class MediaResponse(BaseModel):
    id: str
    title: str
    year: int | None
    media_type: str
    rating: str | None
    poster_url: str | None
    quality: str | None
    size: str | None
    size_bytes: float
    runtime: str | None
    video_codec: str | None
    container: str | None
    file_path: str | None
    audio_tracks: list[AudioTrackResponse]
    subtitle_tracks: list[SubtitleTrackResponse]


class MediaListResponse(BaseModel):
    id: str
    title: str
    year: int | None
    media_type: str
    rating: str | None
    poster_url: str | None
    quality: str | None
    size: str | None
    size_bytes: float
    runtime: str | None
    video_codec: str | None
    container: str | None
    audio_track_count: int
    subtitle_track_count: int


class StatsResponse(BaseModel):
    total_media: int
    total_movies: int
    total_shows: int
    total_audio_tracks: int
    total_subtitle_tracks: int
    total_size_gb: float
